docs/components/select.md

```diff
@@ -1,1013 +1,741 @@
-    <div class="select-item-label">{{ item.label }}</div>
-    <div class="select-item-desc">{{ item.label }}</div>
+  <template #item-template="{ item }">
+    <KongIcon />
+    {{ item?.label }}
   </template>
 </KSelect>
-
-<script>
-function debounce(func, timeout) {
-  let timer;
-  return function (...args) {
-    clearTimeout(timer);
-    timer = setTimeout(() => {
-      func.apply(this, args);
-    }, timeout);
-  };
-}
-const allItems = new Array(10).fill().map((_, i) => ({
-  label: `Item ${i}`,
-  description: `This is the description for item ${i}.`,
-  value: `autosuggest-item-${i}`,
-  ...(i > 5 && { group: `${i % 2 === 0 ? 'Even items greater than 5' : 'Odd items greater than 5'}` })
-}));
-export default {
-  data() {
-    return {
-      defaultItems: [],
-      items: [],
-      loading: true,
-    }
-  },
-  methods: {
-    onQueryChange (val) {
-      if (val === '' && !this.defaultItems.length) {
-        // If query is empty and default items are not fetched, fetch them
-        this.loading = true;
-        setTimeout(() => {
-          this.defaultItems = allItems;
-          this.items = this.defaultItems.map(item => Object.assign({}, item));
-          this.loading = false;
-        }, 200);
-        return;
-      }
-      if (val === '') {
-        // If query is empty and default items are fetched, use the default items
-        this.items = this.defaultItems.map(item => Object.assign({}, item));
-        return;
-      }
-      this.debouncedHandler(val);
-    },
-    debouncedHandler: debounce(function (val) {
-      this.loading = true;
-      // Fetch items that contain the keyword
-      setTimeout(() => {
-        this.items =
-          allItems
-            .filter(item => item.label.toLowerCase().includes(val.toLowerCase()) || item.description.toLowerCase().includes(val.toLowerCase()))
-            .map(item => Object.assign({}, item));
-        this.loading = false;
-      }, 200);
-    }, 400)
-  }
-}
-</script>
 ```
 
-### loading
+### positionFixed
 
-When `autosuggest` is enabled, you can use the `loading` prop to show a loading indicator while fetching data from API.
-By default, the loading indicator is a spinner icon, and you can implement your own indicator using the `loading` slot.
-See [autosuggest](#autosuggest) for an example.
+Use fixed positioning of the popover to avoid content being clipped by parental boundaries. See [KPopover](/components/popover#positionfixed) docs for more information.
 
-### reuseItemTemplate
+### enableItemCreation
 
-Use this prop to customize selected item element appearance by reusing content passed through [item-template](#item-template) slot. **Note:** this prop only applies when `appearance` prop is `select` (use [selected-item-template](#selected-item-template) slot in other cases).
+When used in conjunction with `enableFiltering` set to `true`, KSelect will suggest adding a new value when filtering produces no results.
```

Comment:
issue: we need to sanitize HTML and input to prevent things like this (same comment on the multiselect PR)
![image](https://github.com/Kong/kongponents/assets/2229946/2e285420-1929-4eb5-a712-5dda5aeb49b0)

Reply:
If a user enters HTML or script tags when creating a new item, it's adding that HTML directly into the DOM (XSS).

You need to sanitize the user input to prevent this.

Reply:
Added a quick and easy way to sanitize user input: https://github.com/Kong/kongponents/pull/1832/commits/42cd593c12afc33cf797146d3074c346daf35c59. Though I would argue that this would need to be a separate ticket.

Reply:
This looks good; you'll need this on KMultiselect too